reana_workflow_engine_snakemake/config.py

```diff
@@ -20,3 +21,39 @@
 
 SNAKEMAKE_MAX_PARALLEL_JOBS = 100
 """Snakemake maximum number of jobs that can run in parallel."""
+
+POLL_JOBS_STATUS_SLEEP_IN_SECONDS = 5
+"""Time to sleep between polling for job status."""
+
+
+# defined in reana-db component, in reana_db/models.py file as JobStatus
+class JobStatus(str, Enum):
+    """Enumeration of job statuses.
+
+    Example:
+        JobStatus.started == "started"  # True
+    """
+
+    # FIXME: this state is not defined in reana-db but returned by r-job-controller
+    started = "started"
+
+    created = "created"
+    running = "running"
+    finished = "finished"
+    failed = "failed"
+    stopped = "stopped"
+    queued = "queued"
+
+
+# defined in reana-db component, in reana_db/models.py file as RunStatus
+class RunStatus(Enum):
+    """Enumeration of possible run statuses of a workflow."""
+
+    created = 0
+    running = 1
+    finished = 2
+    failed = 3
+    deleted = 4
+    stopped = 5
+    queued = 6
+    pending = 7
```

Comment:
Thought: I'm not sure if it's a good idea to replicate these enums in all our packages. I know our current solution is not ideal but creating these enums makes things harder to maintain in the future. Would it make sense to have this centralized in `reana-commons`?

Reply:
This is cleaner than the current solution. The previous were local variables with mappings to numbers, duplicated in multiple functions (the same situation was in Yadage and, if not mistaken, currently, CWL).

Regarding maintenance, we do not change those values much so minimal overhead here. In addition, enums are stored in one single file `config.py` so in case of change no need to search in files for values like `1`, `3`, etc.

I am not a fan of adding enums to `reana-commons`. It makes the engine even more dependant ("centralized") on the `r-commons` package. I am not sure what is our high-level strategy here, centralization or independence of cluster components. I assume, if we are advertised as "microservices", it means independence but current architecture is kinda monolithic.

[the thread on one of Yadage PRs on the same topic](https://github.com/reanahub/reana-workflow-engine-yadage/pull/206#discussion_r727121515)

*suggestion:* I am open to discussions and will go along with whatever we agree. But, I would suggest making possible changes in the next PR. You can open an issue and we can discuss what to do.

*Note:* If we will go with centralization, this PR will actually help to achieve this. Both enums are named the same way as in `reana-db` so it will be a matter of moving enums from `reana-db` to `r-commons` and changing "import" statements in the engines.

P.S Beware also of `started` status used by `job-controller` which is not presented in `reana-db`.

Reply:
I see... I read your discussion and the hard feelings involved so I don't want to get into it...

Just my opinion. I don't think that centralizing an enum that can be potentially used in many components is going towards a monolithic architecture... `reana-commons` acts as a util function, the same way as any other python library.

Following your logic, then we should remove the `reana-commons` dependency altogether and keep repeated code all over the engines such as `WorkflowStatusPublisher` and the `JobControllerAPIClient`, is this what you mean?

Reply:
My logic was is that `r-commons` has too much stuff inside: engines-specific code, consumer code, duplications of `job-controller` OpenAPI specification, email, k8s, htcondor, etc. Now, we want to add a bit more stuff to this.

So it is not about duplicating everything everywhere. Some of it can be duplicated (it doesn't mean it is bad), some of it can go to the already existing repo (e.g, if it is used only in this repo already), some of it can stay in the `r-commons`. 

Point is, before adding code to `r-commons` it is good to check if it makes sense. This is an external package and not a `utils` file or folder in a monorepo. New code, a new release of `r-commons`, updating other components, more code, higher chances of changing something in `r-commons`, more releases of `r-commons`, more updates of other components. Of course, duplication has its own problems too.

That is why I suggested creating an issue and discussing `Enum`s there.

Reply:
I created an [issue](https://github.com/reanahub/reana-commons/issues/324) for this.